### PR TITLE
[SPARK-10373] [PYSPARK] move @since into pyspark from sql

### PR DESCRIPTION
**Pull request by davies**
cc @mengxr

### CODE REVIEW

**Comment by rxin** [comment]
Would this also import the since function if somebody does "from pyspark import *"?

**Comment by davies** [comment]
@rxin No, it's not in `__all__`.

File: python/pyspark/__init__.py [hunk 1]
```diff
@@ -48,6 +48,22 @@
...
 
+
+def since(version):
+    """
+    A decorator that annotates a function to append the version of Spark the function was added.
+    """
+    import re
+    indent_p = re.compile(r'\n( +)')
+
+    def deco(f):
```
Comment (noel-smith) [inline python/pyspark/__init__.py]: Would it be OK to add a clause to handle situations where f.__doc__ is None? There are a handful of methods without docstrings. Thanks!

Comment (davies) [inline python/pyspark/__init__.py]: I think it's better to have some docs for public APIs.

Comment (noel-smith) [inline python/pyspark/__init__.py]: You're right, that makes more sense. Ignore the comment above.

**Comment by mengxr** [comment]
LGTM. Merged into master. Thanks!

**Pull request closed without merging**